api/lib/opentelemetry/trace/sampling_hint.rb

```diff
@@ -16,7 +16,7 @@ module SamplingHint
       RECORD = :__record__
 
       # Suggest to record events and propagate.
-      RECORD_AND_PROPAGATE = :__record_and_propagate__
+      RECORD_AND_SAMPLED = :__record_and_propagate__
```

Comment:
Is there any reason not to make the value `:__record_and_sampled__` to match the constant name?